src/api/libcellml/issue.h

```diff
@@ -0,0 +1,426 @@
+     */
+    void setCause(Cause cause);
+
+    /**
+     * @brief Get the cause of this issue.
+     *
+     * Get the @c cause of this issue. If no cause has been set for
+     * this issue, will return Cause::UNDEFINED.
+     *
+     * @return The @c Issue::Cause set for this issue.
+     */
+    Cause cause() const;
+
+    /**
+     * @brief Check whether the cause of this issue matches the argument cause.
+     *
+     * Return @c true if the @p cause matches the the @c Issue::Cause for this
+     * issue and @c false otherwise.
+     *
+     * @return @c true if the @p cause matches the the @c Issue::Cause for this
+     * issue and @c false otherwise.
+     */
+    bool isCause(Cause cause) const;
+
+    /**
+     * @brief Set the level of this issue.
+     *
+     * Set the @p level of this issue from the options available in
+     * @c Issue::Level.
+     *
+     * @param level The @c Issue::Level to set.
+     */
+    void setLevel(Level level);
+
+    /**
+     * @brief Get the level of this issue.
+     *
+     * Get the @c level of this issue. If no level has been set for
+     * this issue, will return Level::ERROR.
+     *
+     * @return The @c Issue::Level set for this issue.
+     */
+    Level level() const;
+
+    /**
+     * @brief Check whether the level of this issue matches the argument level.
+     *
+     * Return @c true if the @p level matches the the @c Issue::Level for this
+     * issue and @c false otherwise.
+     *
+     * @return @c true if the @p level matches the the @c Issue::Level for this
+     * issue and @c false otherwise.
+     */
+    bool isLevel(Level level) const;
+
+    /**
+     * @brief Set the @c enum ReferenceRule of this issue.
+     *
+     * Set the @p rule of this issue from the options available in
+     * the @c ReferenceRule @c enum.
+     *
+     * @param rule The @c ReferenceRule to set.
+     */
+    void setRule(ReferenceRule rule);
+
+    /**
+     * @brief Get the @c enum ReferenceRule of this issue.
+     *
+     * Get the @c enum @c ReferenceRule value @p rule of this issue. If
+     * no rule has been set for this issue, will return ReferenceRule::UNDEFINED.
+     *
+     * @return The @c ReferenceRule for this issue.
+     */
+    ReferenceRule rule() const;
+
+    /**
+     * @brief Get the @c url of this issue.
+     *
+     * Get the @c url to which the user should refer for more information on this issue.
+     *
+     * @return The @c url for this issue.
+     */
+    std::string url() const;
+
+    /**
+     * @brief Get the @c std::string heading associated with the @c enum ReferenceRule for this issue.
+     *
+     * Get the @c std::string CellML 2.0 Reference heading associated with the @c enum ReferenceRule
+     * for this issue. If no rule has been set for this issue, will return an empty string.
+     *
+     * @return The @c std::string referencing the CellML 2.0 Reference heading relevant to this issue.
+     */
+    std::string referenceHeading() const;
```

Comment:
Talks of the `CellML 2.0 Reference` but the reference heading is able to have items from code generation which has nothing to do with CellML?

Reply:
What would you like it to be called?  The reference document(s) that would contain all the specification stuff as well as any other "rules" type things for code generation, imports etc?  I could just remove the CellML 2.0 if that's the part that's problematic?

Reply:
I see this as a libCellML reference document.

Reply:
OK, I've changed it now.